File: Leetcode/JS/Leetcode419-甲板上的战舰.js
var countBattleships = function(board) {
    let ans = 0,n = board.length,m = board[0].length;
    for(let i = 0;i < n;i++) {
        for(let j = 0;j < m;j++) {
            if(i > 0 && board[i - 1][j] == 'X') continue;
            if(j > 0 && board[i][j - 1] == 'X') continue;
            if(board[i][j] == 'X') ans++;
        }
    }
    return ans;
};

File: Leetcode/PY/Leetcode419-甲板上的战舰.py
class Solution:
    def countBattleships(self, board: List[List[str]]) -> int:
        n,m,ans = len(board),len(board[0]),0
        for i in range(n):
            for j in range(m):
                if i > 0 and board[i - 1][j] == 'X': continue
                if j > 0 and board[i][j - 1] == 'X': continue
                if board[i][j] == 'X': ans += 1
        return ans 